pkg/resources/validation/cluster/cluster_test.go

```diff
@@ -0,0 +1,101 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/webhook/pkg/admission"
+	"github.com/stretchr/testify/assert"
+	admissionv1 "k8s.io/api/admission/v1"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	v1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+type mockReviewer struct {
+	v1.SubjectAccessReviewExpansion
+}
+
+func (m *mockReviewer) Create(
+	_ context.Context,
+	_ *authorizationv1.SubjectAccessReview,
+	_ metav1.CreateOptions,
+) (*authorizationv1.SubjectAccessReview, error) {
+	return &authorizationv1.SubjectAccessReview{
+		Status: authorizationv1.SubjectAccessReviewStatus{
+			Allowed: true,
+		},
+	}, nil
+}
+
+type testCase struct {
+	oldCluster     v3.Cluster
+	newCluster     v3.Cluster
+	operation      admissionv1.Operation
+	expectAllowed  bool
+	expectedReason metav1.StatusReason
+}
+
+func TestAdmit(t *testing.T) {
+	for name, tc := range map[string]testCase{
+		"Create": {
+			operation:     admissionv1.Create,
+			expectAllowed: true,
+		},
+		"UpdateWithUnsetFleetWorkspaceName": {
+			oldCluster:     v3.Cluster{Spec: v3.ClusterSpec{FleetWorkspaceName: "fleet-default"}},
+			operation:      admissionv1.Update,
+			expectAllowed:  false,
+			expectedReason: metav1.StatusReasonInvalid,
+		},
+		"UpdateWithNewFleetWorkspaceName": {
+			oldCluster:    v3.Cluster{Spec: v3.ClusterSpec{FleetWorkspaceName: "fleet-default"}},
+			newCluster:    v3.Cluster{Spec: v3.ClusterSpec{FleetWorkspaceName: "new"}},
+			operation:     admissionv1.Update,
+			expectAllowed: true,
+		},
+		"UpdateWithUnchangedFleetWorkspaceName": {
+			oldCluster:    v3.Cluster{Spec: v3.ClusterSpec{FleetWorkspaceName: "fleet-default"}},
+			newCluster:    v3.Cluster{Spec: v3.ClusterSpec{FleetWorkspaceName: "fleet-default"}},
+			operation:     admissionv1.Update,
+			expectAllowed: true,
+		},
+		"Delete": {
+			oldCluster:    v3.Cluster{Spec: v3.ClusterSpec{FleetWorkspaceName: "fleet-default"}},
+			operation:     admissionv1.Delete,
+			expectAllowed: true,
+		},
+	} {
```

Comment:
For table-driven tests, the standard format is first to define a slice of tests and then loop over that slice. There are some examples in our code, but here is an article to reference as well, Though you don't need to define an anonymous struct the like the example.

https://dave.cheney.net/2019/05/07/prefer-table-driven-tests
``` golang
func TestSplit(t *testing.T) {
    tests := []struct {
        input string
        sep   string
        want  []string
    }{
        {input: "a/b/c", sep: "/", want: []string{"a", "b", "c"}},
        {input: "a/b/c", sep: ",", want: []string{"a/b/c"}},
        {input: "abc", sep: "/", want: []string{"abc"}},
    } 

    for _, tc := range tests {
        got := Split(tc.input, tc.sep)
        if !reflect.DeepEqual(tc.want, got) {
            t.Fatalf("expected: %v, got: %v", tc.want, got)
        }
    }
}
```